Detectors/TRD/base/include/TRDBase/TRDDiffAndTimeStructEstimator.h

```diff
@@ -0,0 +1,60 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+#ifndef O2_TRDDIFFANDTIMESTRUCTESTIMATOR_H
+#define O2_TRDDIFFANDTIMESTRUCTESTIMATOR_H
+
+#include <array>
+
+namespace o2
+{
+namespace trd
+{
+
+// CONTANT TIME STRUCTURE DATA FROM GARFIELD
+constexpr int ktimebin = 38;
+constexpr int kZbin = 11;
+
+/// A class to calculate diffusion and time structure values (GARFIELD model)
+/// (used in digitization). Class was split off trom TRDCommonParam
+/// and is no longer a singleton so that we can use it in a multithreaded context.
+class TRDDiffusionAndTimeStructEstimator
+{
+ public:
+  TRDDiffusionAndTimeStructEstimator() = default;
+
+  /// determines the diffusion coefficients as a function of drift velocity
+  bool GetDiffCoeff(float& dl, float& dt, float vdrift);
+
+  /// determines drift time as function of drift velocity and coordinates
+  float TimeStruct(float vdrift, float xd, float z);
+
+ private:
+  void SampleTimeStruct(float vdrift);
+
+  std::array<float, ktimebin * kZbin> mTimeStruct1; //! cached Time Structure of Drift Cells (for last vdrift value)
```

Comment:
same. Can be optimized later.

Reply:
(I was actually planning to avoid the copy but wanted to switch to using to the new classes but based on the same code first)

Reply:
OK, for the record, adding a comment which was lost:

https://github.com/AliceO2Group/AliceO2/blob/f69f68aa223621b69a43d0771bff43aadd84eac5/Detectors/TRD/base/src/TRDDiffAndTimeStructEstimator.cxx#L544
The ``ib`` calculation is likely to be wrong, also in its AliRoot prototype since it maps 0<b<5 kGauss field on ``0<=ib<=4`` but maps ``-5<b<0`` field to ib=0. I think the TRD experts (@tdietel or C.Lippmann, the author of the original code) should clarify what is the supposed mapping.